src/hooks.ts

```diff
@@ -201,38 +232,57 @@ export const useExperiment: UseExperiment = (experimentKey, options = {}, overri
  *       ClientReady and DidTimeout provide signals to handle this scenario.
  */
 export const useFeature: UseFeature = (featureKey, options = {}, overrides = {}) => {
-  const { isServerSide, optimizely, timeout } = useContext(OptimizelyContext);
+  const { optimizely, isServerSide } = useContext(OptimizelyContext);
   if (!optimizely) {
     throw new Error('optimizely prop must be supplied via a parent <OptimizelyProvider>');
   }
 
-  // Helper function to return the current values for isEnabled and variables.
-  const getCurrentValues = useCallback<() => FeatureDecisionValues>(
+  const overrideAttrs = useCompareAttrsMemoize(overrides.overrideAttributes);
+  const getCurrentDecision: () => FeatureDecisionValues = useCallback(
     () => ({
-      isEnabled: optimizely.isFeatureEnabled(featureKey, overrides.overrideUserId, overrides.overrideAttributes),
-      variables: optimizely.getFeatureVariables(featureKey, overrides.overrideUserId, overrides.overrideAttributes),
+      isEnabled: optimizely.isFeatureEnabled(featureKey, overrides.overrideUserId, overrideAttrs),
+      variables: optimizely.getFeatureVariables(featureKey, overrides.overrideUserId, overrideAttrs),
     }),
-    [featureKey, overrides]
+    [optimizely, featureKey, overrides.overrideUserId, overrideAttrs]
   );
 
-  // Set the initial state immediately serverSide
-  const [state, setState] = useState<UseFeatureState>(() => {
-    const initialState = {
-      isEnabled: false,
-      variables: {},
-      clientReady: isServerSide ? true : false,
-      didTimeout: false,
-    };
-    if (isServerSide) {
-      return { ...initialState, ...getCurrentValues() };
+  const isClientReady = isServerSide || optimizely.isReady();
+  const [decisionState, setDecisionState] = useState<FeatureDecisionValues>(() => {
+    if (isClientReady) {
+      return getCurrentDecision();
     }
-    return initialState;
+    return { isEnabled: false, variables: {} };
```

Comment:
I am guessing there is something about this implementation of `useState` that I don’t understand. Can you help me understand what the anon function does here - does it only set initial state or is it used on every `setDecisionState`. I look at the React docs but couldn't find a clear example. The one I found seemed to insinuate that the provided function would be called every time.